ocs_ci/ocs/ui/page_objects/block_and_file.py

```diff
@@ -181,3 +183,53 @@ def get_raw_capacity_card_values(self):
         )
 
         return used, available
+
+    def get_estimated_days_from_consumption_trend(self):
+        """
+        This will get the web element xpaths from views.py and try to get text from those elements
+
+        This will fetch information from DataFoundation>>Storage>>Block and File page>>Consumption trend card
+
+        Returns:
+            tuple: (get_est_days_from_element, get_avg_from_element)
+
+        """
+        get_est_days_from_element = self.get_element_text(
+            self.validation_loc["locate_estimated_days_along_with_value"]
+        )
+        get_avg_from_element = self.get_element_text(
+            self.validation_loc["locate_average_of_storage_consumption"]
+        )
+        return (get_est_days_from_element, get_avg_from_element)
+
+    def odf_storagesystems_consumption_trend(self):
+        """
+        Function to verify changes and validate elements on ODF storage consumption trend for ODF 4.17
+        This will navigate through below order
+        DataFoundation>>Storage>>storagecluster_storagesystem_details>>Block and File page
+        Further it looks for the Consumption trend card
+
+        Returns:
+            tuple: tpl_of_days_and_avg  ex: (Estimated days, Average)
+
+        """
+        ValidationUI().odf_console_plugin_check()
+        storage_systems_page = (
+            ValidationUI().nav_odf_default_page().nav_storage_systems_tab()
+        )
+        # PageNavigator().nav_odf_default_page().nav_storage_systems_tab()
+        storage_system_details = (
+            storage_systems_page.nav_storagecluster_storagesystem_details()
+        )
+        storage_system_details.nav_details_overview()
+        storage_system_details.nav_details_object()
```

Comment:
1. Navigation to storage_system_tab and then to nav_details_object() and then back to nav_block_and_file() has no sense. It will just burn execution time. 
2.  All the navigation should happen within the test method but not in the BlockAndFile class. You supposed to use methods from this class being on that page already (Page Object Model pattern). 

This approach calls Fluent Interface Design Pattern - https://www.lambdatest.com/blog/fluent-interface-design-pattern/ 

and 

Page Object Model pattern -https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/
----
Please move this 

```
block_and_file_page = PageNavigator()
        .nav_odf_default_page()
        .nav_storage_systems_tab()
        .nav_storagecluster_storagesystem_details()
        .nav_block_and_file()
consumption_trend = block_and_file_page.odf_storagesystems_consumption_trend()
```
it will be clean and clear for everyone. 

your method `odf_storagesystems_consumption_trend` should start from:
`if not config.ENV_DATA["mcg_only_deployment"]`

Reply:
I fetched your PR, updated it on my branch.

all smooth, passed 2 times -> https://privatebin.corp.redhat.com/?625372707e17b576#E9qcCeLFUn1HxBExnLXjzg2U5ebup78R9JKMBFuTPbXT

cherry-pick this https://github.com/DanielOsypenko/ocs-ci/tree/pr-10520